crates/uv/src/cli.rs

```diff
@@ -1611,6 +1611,10 @@ pub(crate) struct AddArgs {
     #[arg(long)]
     pub(crate) dev: bool,
 
+    /// Add the requirements as workspace dependencies.
+    #[arg(long)]
+    pub(crate) workspace: bool,
```

Comment:
I'm on the fence whether we need an explicit argument for this, we could also do workspace discovery and know that it has to be a workspace dep, otoh it's consistent with us requiring `workspace = true` in `pyproject.toml` (it's something we took from cargo, it's helpful to make we don't accidentally flip between workspace and index and to keep the information about the package source local)